pyproject.toml

```diff
@@ -37,13 +37,12 @@ requires-python = ">=3.10"
 
 [project.optional-dependencies]
 lint = [
-    "black==24.10.0",
-    "codespell[toml]==2.3.0",
-    "yamllint==1.35.1"
+    "codespell[toml]~=2.3",
+    "yamllint~=1.35"
 ]
 types = [
-    "mypy[reports]==1.13.0",
-    "pyright==1.1.384",
+    "mypy[reports]~=1.13",
+    "pyright==1.1.390",
```

Comment:
I would prefer to take the opportunity and drop the pyright dep from PyPI and just use the snap

Reply:
This is probably the best opportunity, so done. 

It needed a `common.mk` update so I've applied this to starbase too: https://github.com/canonical/starbase/pull/299